client/src/components/client-area/ProgressOrder.vue

```diff
@@ -0,0 +1,313 @@
+          <ion-text class="body">{{ $msTranslate('clientArea.dashboard.step.cancel.description') }}</ion-text>
+        </div>
+      </ms-report-text>
+      <div
+        class="process-step"
+        v-for="(step, index) in steps"
+        :key="index"
+        :class="{
+          'process-step-todo': getStep(customOrderBmsStatus, customOrderSellsyStatus) < getStep(step.statusBms, step.statusRequest),
+          'process-step-active': getStep(customOrderBmsStatus, customOrderSellsyStatus) === getStep(step.statusBms, step.statusRequest),
+          'process-step-done': getStep(customOrderBmsStatus, customOrderSellsyStatus) > getStep(step.statusBms, step.statusRequest),
+        }"
+      >
+        <div class="process-step-icon">
+          <div
+            class="dot-todo"
+            v-if="getStep(customOrderBmsStatus, customOrderSellsyStatus) < getStep(step.statusBms, step.statusRequest)"
+          />
+          <div
+            class="dot-active"
+            v-if="getStep(customOrderBmsStatus, customOrderSellsyStatus) === getStep(step.statusBms, step.statusRequest)"
+          />
+          <ion-icon
+            class="process-step-icon__item"
+            :icon="checkmarkCircle"
+            v-if="getStep(customOrderBmsStatus, customOrderSellsyStatus) > getStep(step.statusBms, step.statusRequest)"
+          />
+        </div>
+        <div class="process-step-text">
+          <ion-text class="process-step-text__title title-h3">
+            {{ $msTranslate(step.title) }}
+          </ion-text>
+          <ion-text
+            class="process-step-text__info body"
+            v-if="getStep(customOrderBmsStatus, customOrderSellsyStatus) === getStep(step.statusBms, step.statusRequest)"
+          >
+            {{ $msTranslate(step.description) }}
+          </ion-text>
+        </div>
+      </div>
+    </div>
+  </template>
+  <template v-else-if="querying">
+    <ms-spinner />
+  </template>
+  <template v-else-if="error">
+    <ms-report-text :theme="MsReportTheme.Error">
+      {{ $msTranslate(error) }}
+    </ms-report-text>
+  </template>
+</template>
+
+<script setup lang="ts">
+import { IonIcon, IonText } from '@ionic/vue';
+import { checkmarkCircle } from 'ionicons/icons';
+import { BmsAccessInstance, DataType, BmsOrganization, CustomOrderStatus, CustomOrderRequestStatus } from '@/services/bms';
+import { isDefaultOrganization } from '@/views/client-area/types';
+import { ref, onMounted } from 'vue';
+import { MsSpinner, MsReportText, MsReportTheme } from 'megashark-lib';
+import { getCustomOrderStatusTranslationKey } from '@/services/translation';
+
+const customOrderBmsStatus = ref<CustomOrderStatus>(CustomOrderStatus.Unknown);
+const customOrderSellsyStatus = ref<CustomOrderRequestStatus>(CustomOrderRequestStatus.Received);
+
+const querying = ref(true);
+const error = ref('');
+
+const props = defineProps<{
+  organization: BmsOrganization;
+}>();
+
+const steps = [
+  {
```

Comment:
You can do a pro-gamer move with
```
{
  ...getCustomOrderStatusTranslationKey(CustomOrderStatus.NothingLinked, CustomOrderRequestStatus.Received),
  id: 1,
  statusBms: CustomOrderStatus.NothingLinked,
  statusRequest: CustomOrderRequestStatus.Received,
}
```
It will auto-assign `title` and `description`, save you a useless function call, and a few lines overall.